lite/examples/sound_classification_use_task_api/ios/SoundClassification/ViewController.swift

```diff
@@ -0,0 +1,153 @@
+// Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import AVFoundation
+import TensorFlowLiteTaskAudio
+
+class ViewController: UIViewController {
+
+  // MARK: - Variables
+  @IBOutlet weak var tableView: UITableView!
+  @IBOutlet weak var inferenceView: InferenceView!
+
+  private var datas: [ClassificationCategory] = []
+
+  private var modelType: ModelType = .Yamnet
+  private var overLap: Double = 0.5
+  private var maxResults: Int = 3
+  private var threshold: Float = 0.3
+  private var threadCount: Int = 2
+
+  private var timer: Timer?
+
+  // MARK: - View controller lifecycle methods
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    inferenceView.setDefault(model: modelType, overLab: overLap, maxResult: maxResults, threshold: threshold, threads: threadCount)
+    inferenceView.delegate = self
+    startAudioRecognition()
+  }
+
+  // MARK: - Private Methods
+  /// Request permission and start audio classification if granted
+  private func startAudioRecognition() {
+    AVAudioSession.sharedInstance().requestRecordPermission { granted in
+      if granted {
+        self.classification()
+      } else {
+        self.checkPermissions()
+      }
+    }
+  }
+
+  /// Check permission and show error if user denied permission
+  private func checkPermissions() {
+    switch AVAudioSession.sharedInstance().recordPermission {
+    case .granted, .undetermined:
+      startAudioRecognition()
+    case .denied:
+      showPermissionsErrorAlert()
+    @unknown default:
+      fatalError()
+    }
+  }
+
+  /// Run audio classification
+  private func classification() {
+    guard let soundClassificationHelper = SoundClassificationHelper(
+      modelType: modelType,
+      threadCount: threadCount,
+      scoreThreshold: threshold,
+      maxResults: maxResults) else { return }
+    soundClassificationHelper.delegate = self
```

Comment:
There's a bug that when you switch back and forth between Yamnet and speech command, a new helper instance is created and all of them will continue running all the time.

Once the user has changed the model, the old helper should stop running.